pkg/endpoints/pipeline.go

```diff
@@ -166,6 +212,112 @@ func (r Resource) getPipelineRun(request *restful.Request, response *restful.Res
 	response.WriteEntity(pipelinerun)
 }
 
+func (r Resource) createPipelineRun(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter("namespace")
+
+	pipelineRunData := ManualPipelineRun{}
+	if err := request.ReadEntity(&pipelineRunData); err != nil {
+		logging.Log.Error("error parsing request body on call to createPipelineRun %s", err)
+		utils.RespondError(response, err, http.StatusBadRequest)
+		return
+	}
+
+	createResponse := r.CreatePipelineRunImpl(pipelineRunData, namespace)
+	if createResponse.CODE == 204 {
+		response.WriteHeader(http.StatusNoContent)
+	} else { // anything other than 204 is an error - RespondError
+		utils.RespondError(response, createResponse.ERROR, createResponse.CODE)
+	}
+}
+
+// CreatePipelineRunImpl ...
+/* Create a new manual pipeline run and resources in a given namespace */
+func (r Resource) CreatePipelineRunImpl(pipelineRunData ManualPipelineRun, namespace string) *AppError {
+
+	pipelineName := pipelineRunData.PIPELINENAME
+
+	saName := ""
+	providedSaName := os.Getenv("PIPELINE_RUN_SERVICE_ACCOUNT")
+	if providedSaName != "" {
+		saName = providedSaName
+	} else {
+		saName = "default"
+	}
+
+	logging.Log.Infof("PipelineRuns will be created in the namespace %s", namespace)
+	logging.Log.Infof("PipelineRuns will be created with the service account %s", saName)
+
+	startTime := getDateTimeAsString()
+
+	// Assumes you've already applied the yml: so the pipeline definition and its tasks must exist upfront.
+	generatedPipelineRunName := fmt.Sprintf("devops-pipeline-run-%s", startTime)
+
+	pipeline, err := r.getPipelineImpl(pipelineName, namespace)
+	// getPipelineImpl returns a null pipeline object if it doesn't exist, so be safe and check for that
+	if err != nil || pipeline.Name == "" {
```

Comment:
I don't think you need to check if the `pipeline.Name == ""` here because `getPipelineImpl` returns an error if the pipeline is not found (https://github.com/tektoncd/dashboard/pull/33/files#diff-2a268056787157d73c4168498850de86R151).

Also, you can remove the  comment `// getPipelineImpl returns a null pipeline object if it doesn't exist, so be safe and check for that` because you changed the `getPipelineImpl` to return `v1alpha1.Pipeline{}` instead of nil.

Reply:
true, removing both